modules/objdetect/test/test_qrcode_encode.cpp

```diff
@@ -450,6 +450,34 @@ TEST(Objdetect_QRCode_Encode_Decode_Structured_Append, DISABLED_regression)
 
 #endif // UPDATE_QRCODE_TEST_DATA
 
+TEST(Objdetect_QRCode_Encode_Decode_Structured_Append, regression_22205)
+{
+    cv::Ptr<cv::QRCodeEncoder> encoder = nullptr;
+    std::string input_data = "the quick brown fox jumps over the lazy dog";
+
+    std::vector<cv::QRCodeEncoder::EncodeMode> encode_modes = {
+        cv::QRCodeEncoder::EncodeMode::MODE_NUMERIC,
+        cv::QRCodeEncoder::EncodeMode::MODE_ALPHANUMERIC,
+        cv::QRCodeEncoder::EncodeMode::MODE_BYTE};
```

Comment:
It makes sense to use parameterized test approach. It's move verbose and easy to debug in case of failure. See https://github.com/opencv/opencv/blob/4.x/modules/calib3d/test/test_affine2d_estimator.cpp#L46 as example and https://github.com/google/googletest/blob/main/docs/advanced.md#value-parameterized-tests for documentation.